src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,65 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+interface Props {
+  todo: Todo;
+  onSelect?: (todo: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, onSelect }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(todo.userId).then(fetchedUser => {
+      setUser(fetchedUser);
+      setLoading(false);
+    });
```

Comment:
Ensure that the `getUser` function returns a promise, as the code assumes it does by using `.then()`. If it doesn't, you may need to adjust the implementation of `getUser`.